### PR TITLE
Add resource type internet_gateway

### PR DESCRIPTION
**Pull request by inokappa**
Hi, Koyama san

I added Resource type `internet_gateway`.
And I think that it is related to issue #216.

### exist

```ruby
describe internet_gateway('igw-1ab2cd3e') do
  it { should exist }
end

describe internet_gateway('my-internet-gateway') do
  it { should exist }
end
```

### be_attached_to

```ruby
describe internet_gateway('igw-1ab2cd3e') do
  it { should be_attached_to('vpc-ab123cde') }
end
```

### have_tag

```ruby
describe internet_gateway('igw-1ab2cd3e') do
  it { should have_tag('Name').value('my-internet-gateway') }
end
```

Please confirm. 🙏 

Regards.

Yohei.

### CODE REVIEW

File: lib/awspec/helper/finder/internet_gateway.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module Awspec::Helper
+  module Finder
+    module InternetGateway
+      def find_internet_gateway(id)
```
Comment (k1LoW) [inline lib/awspec/helper/finder/internet_gateway.rb]: Use https://github.com/k1LoW/awspec/blob/master/lib/awspec/helper/finder/ec2.rb#L38

File: lib/awspec/type/internet_gateway.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    def resource_via_client
+      @resource_via_client ||= find_internet_gateway(@display_name)
+    end
+
+    def id
+      @id ||= resource_via_client.internet_gateway_id if resource_via_client
+    end
+
+    def attached_to?(vpc_id)
```
Comment (k1LoW) [inline lib/awspec/type/internet_gateway.rb]: Could you support `tag:Name` ( You can use `find_vpc()` ) ?

File: lib/awspec/type/internet_gateway.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      resource_via_client.attachments.first.vpc_id == vpc_id && \
```
Comment (k1LoW) [inline lib/awspec/type/internet_gateway.rb]: Could you support multi attachments?

```ruby
resource_via_client.attachments.find do |a|
    a.vpc_id == vpc_id && a.state == 'attached'
end
```

**Comment by k1LoW** [comment]
Hi @inokappa !!! Nice work! Thank you!!

Please check my comment 👍

**Comment by inokappa** [comment]
Hi @k1LoW !!!

Thank you for your advice.
I've learnt a lot from you! Thank you so much!!!
(めっちゃ勉強になっていますmm。有難うございます！)

**Comment by inokappa** [comment]
Hi @k1LoW 

I updated it with reference to your advice.
Please confirm.

Regards.

Yohei

**Comment by k1LoW** [comment]
!!! @inokappa !!! Thank you for your good commit !!
LGTM!!!

**Comment by k1LoW** [comment]
Released as v0.68.0.

**Pull request merged**